SignalMessaging/utils/OWSContactAvatarBuilder.m

```diff
@@ -121,19 +129,19 @@ - (UIImage *)buildDefaultImage
         [initials appendString:@"#"];
     }
 
-    CGFloat fontSize = (CGFloat)self.diameter / 2.8f;
-    
-    UIImage *image = [[JSQMessagesAvatarImageFactory avatarImageWithUserInitials:initials
-                                                                 backgroundColor:self.color
-                                                                       textColor:[UIColor whiteColor]
-                                                                            font:[UIFont ows_boldFontWithSize:fontSize]
-                                                                        diameter:self.diameter] avatarImage];
+    UIColor *color = [UIColor ows_conversationColorForColorName:self.colorName isShaded:Theme.isDarkThemeEnabled];
+    OWSAssertDebug(color);
 
-    [self.contactsManager.avatarCache setImage:image forKey:self.signalId diameter:self.diameter];
+    UIImage *_Nullable image =
+        [OWSAvatarBuilder avatarImageWithInitials:initials backgroundColor:color diameter:self.diameter];
+    if (!image) {
+        return nil;
```

Comment:
Do we ever expect this to actually be nil? or is this just in line with the CGImage drawing API?

Reply:
It shouldn't ever be nil.

Reply:
OWSFailDebug then? If nothing else it's good documentation.

Reply:
Done.